lib/withOnyx.js

```diff
@@ -28,37 +29,51 @@ export default function (mapOnyxToState) {
                 this.activeConnectionIDs = {};
 
-                const cachedState = _.reduce(mapOnyxToState, (resultObj, mapping, propertyName) => {
-                    const key = Str.result(mapping.key, props);
-                    const value = Onyx.tryGetCachedValue(key, mapping);
-
-                    /**
-                     * If we have a pending merge for a key it could mean that data is being set via Onyx.merge() and someone expects a component to have this data immediately.
-                     *
-                     * @example
-                     *
-                     * Onyx.merge('report_123', value);
-                     * Navigation.navigate(route); // Where "route" expects the "value" to be available immediately once rendered.
-                     *
-                     * In reality, Onyx.merge() will only update the subscriber after all merges have been batched and the previous value is retrieved via a get() (returns a promise).
-                     * So, we won't use the cache optimization here as it will lead us to arbitrarily defer various actions in the application code.
-                     */
-                    if (value !== undefined && !Onyx.hasPendingMergeForKey(key)) {
-                        // eslint-disable-next-line no-param-reassign
-                        resultObj[propertyName] = value;
-                    }
+                const cachedState = _.reduce(
+                    mapOnyxToState,
+                    (resultObj, mapping, propertyName) => {
+                        const key = Str.result(mapping.key, props);
+                        let value = Onyx.tryGetCachedValue(key, mapping);
+                        if (!value && mapping.initialValue) {
+                            value = mapping.initialValue;
+                        }
 
-                    return resultObj;
-                }, {});
+                        /**
+                         * If we have a pending merge for a key it could mean that data is being set via Onyx.merge() and someone expects a component to have this data immediately.
+                         *
+                         * @example
+                         *
+                         * Onyx.merge('report_123', value);
+                         * Navigation.navigate(route); // Where "route" expects the "value" to be available immediately once rendered.
+                         *
+                         * In reality, Onyx.merge() will only update the subscriber after all merges have been batched and the previous value is retrieved via a get() (returns a promise).
+                         * So, we won't use the cache optimization here as it will lead us to arbitrarily defer various actions in the application code.
+                         */
+                        if (
+                            (value !== undefined
+                                && !Onyx.hasPendingMergeForKey(key))
+                            || mapping.allowStaleData
```

Comment:
tbh, I don't know. @Szymon20000 you knew better how it behaves.

Reply:
Looks, like this is the only place where it is used. Seems like it allows the component to be feed undefined data or stale data when there is pending merges. There is no documentation since it is not part of `onyx.js` which is the only file the generator looks at. That's why I manually added a section to the README.